<launch>
    <!-- ros2 launch l3cam_ros2 l3cam_launch.xml "rviz2:=true" "rqt_reconfigure:=true" -->
    <arg name="stream" default="true" />
    <arg name="rviz2" default="false" />
    <arg name="configure" default="true" />
    <arg name="rqt_reconfigure" default="false" />

    <!-- Network configuration -->
    <arg name="ip_address" default="192.168.1.250" />
    <arg name="netmask" default="255.255.255.0" />
    <arg name="gateway" default="0.0.0.0" />
    <arg name="dhcp" default="false" />
    <arg name="local_address" default="" />
    <arg name="device_address" default="" />

    <!-- Pointcloud configuration -->
    <arg name="pointcloud_color" default="0" /> <!-- see pointCloudColor -->
    <arg name="pointcloud_color_range_minimum" default="0" /> <!-- 0 - 400000 -->
    <arg name="pointcloud_color_range_maximum" default="400000" /> <!-- 0 - 400000 -->
    <arg name="distance_range_minimum" default="0" /> <!-- 0 - 400000 -->
    <arg name="distance_range_maximum" default="400000" /> <!-- 0 - 400000 -->
    <arg name="auto_bias" default="true" />
    <arg name="bias_value_right" default="1580" /> <!-- 700 - 3500 -->
    <arg name="bias_value_left" default="1380" /> <!-- 700 - 3500 -->

    <!-- Polarimetric configuration -->
    <arg name="polarimetric_camera_brightness" default="127" /> <!-- 0 - 255 -->
    <arg name="polarimetric_camera_black_level" default="6.0" /> <!-- 0 - 12.5 -->
    <arg name="polarimetric_camera_auto_gain" default="true" />
    <arg name="polarimetric_camera_auto_gain_range_minimum" default="0.0" /> <!-- 0 - 48 -->
    <arg name="polarimetric_camera_auto_gain_range_maximum" default="48.0" /> <!-- 0 - 48 -->
    <arg name="polarimetric_camera_gain" default="24.0" /> <!-- 0 - 48 -->
    <arg name="polarimetric_camera_auto_exposure_time" default="true" />
    <arg name="polarimetric_camera_auto_exposure_time_range_minimum" default="33.5" /> <!-- 33.5 - 66470.6 -->
    <arg name="polarimetric_camera_auto_exposure_time_range_maximum" default="66470.6" /> <!-- 33.5 - 66470.6 -->
    <arg name="polarimetric_camera_exposure_time" default="33.5" /> <!-- 33.5 - 66470.6 -->

    <!-- RGB configuration -->
    <arg name="rgb_camera_brightness" default="0" /> <!-- -15 - 15 -->
    <arg name="rgb_camera_contrast" default="10" /> <!-- 0 - 30 -->
    <arg name="rgb_camera_saturation" default="16" /> <!-- 0 - 60 -->
    <arg name="rgb_camera_sharpness" default="16" /> <!-- 0 - 127 -->
    <arg name="rgb_camera_gamma" default="220" /> <!-- 40 - 500 -->
    <arg name="rgb_camera_gain" default="0" /> <!-- 0 - 63 -->
    <arg name="rgb_camera_auto_white_balance" default="true" />
    <arg name="rgb_camera_white_balance" default="5000" /> <!-- 1000 - 10000 -->
    <arg name="rgb_camera_auto_exposure_time" default="true" />
    <arg name="rgb_camera_exposure_time" default="156" /> <!-- 1 - 10000 -->
    <arg name="rgb_camera_resolution" default="3" /> <!-- see econResolutions -->
    <arg name="rgb_camera_framerate" default="10" /> <!-- 1 - 16 -->

    <!-- Thermal configuration -->
    <arg name="thermal_camera_colormap" default="1" /> <!-- see thermalTypes -->
    <arg name="thermal_camera_temperature_filter" default="false" />
    <arg name="thermal_camera_temperature_filter_min" default="0" /> <!-- -40 - 200 -->
    <arg name="thermal_camera_temperature_filter_max" default="50" /> <!-- -40 - 200 -->

    <!-- Allied Wide configuration -->
    <arg name="allied_wide_camera_black_level" default="0" /> <!-- 0 - 4095 -->
    <arg name="allied_wide_camera_exposure_time" default="4992.4" /> <!-- 63 - 10000000 -->
    <arg name="allied_wide_camera_auto_exposure_time" default="false" />
    <arg name="allied_wide_camera_auto_exposure_time_range_min" default="87.6" /> <!-- 63.1 - 8999990 -->
    <arg name="allied_wide_camera_auto_exposure_time_range_max" default="8999990.0" /> <!-- 87.6 - 10000000 -->
    <arg name="allied_wide_camera_gain" default="0" /> <!-- 0 - 48 -->
    <arg name="allied_wide_camera_auto_gain" default="false" />
    <arg name="allied_wide_camera_auto_gain_range_min" default="0" /> <!-- 0 - 48 -->
    <arg name="allied_wide_camera_auto_gain_range_max" default="48" /> <!-- 0 - 48 -->
    <arg name="allied_wide_camera_gamma" default="1" /> <!-- 0.4 - 2.4 -->
    <arg name="allied_wide_camera_saturation" default="1" /> <!-- 0 - 2 -->
    <arg name="allied_wide_camera_sharpness" default="0" /> <!-- -12 - 12 -->
    <arg name="allied_wide_camera_hue" default="0" /> <!-- -40 - 40 -->
    <arg name="allied_wide_camera_intensity_auto_precedence" default="0" /> <!-- 0(MinimizeNoise) or 1(MinimizeBlur) -->
    <arg name="allied_wide_camera_auto_white_balance" default="false" />
    <arg name="allied_wide_camera_balance_ratio_selector" default="0" /> <!-- 0(Red), 1(Blue) -->
    <arg name="allied_wide_camera_balance_ratio" default="2.4" /> <!-- 0 - 8 -->
    <arg name="allied_wide_camera_balance_white_auto_rate" default="100" /> <!-- 0 - 100 -->
    <arg name="allied_wide_camera_balance_white_auto_tolerance" default="5" /> <!-- 0 - 50 -->
    <arg name="allied_wide_camera_auto_mode_region_height" default="1028" /> <!-- 0 - 1028 -->
    <arg name="allied_wide_camera_auto_mode_region_width" default="1232" /> <!-- 0 - 1232 -->
    <arg name="allied_wide_camera_intensity_controller_region" default="0" /> <!-- 0(AutoMode), 4(FullImage) -->
    <arg name="allied_wide_camera_intensity_controller_target" default="50" /> <!-- 10 - 90 -->
    <arg name="allied_wide_camera_max_driver_buffers_count" default="64" /> <!-- 1 - 4096 -->

    <!-- Allied Narrow configuration -->
    <arg name="allied_narrow_camera_black_level" default="0" /> <!-- 0 - 4095 -->
    <arg name="allied_narrow_camera_exposure_time" default="4992.4" /> <!-- 63 - 10000000 -->
    <arg name="allied_narrow_camera_auto_exposure_time" default="false" />
    <arg name="allied_narrow_camera_auto_exposure_time_range_min" default="87.6" /> <!-- 63.1 - 8999990 -->
    <arg name="allied_narrow_camera_auto_exposure_time_range_max" default="8999990.0" /> <!-- 87.6 - 10000000 -->
    <arg name="allied_narrow_camera_gain" default="0" /> <!-- 0 - 48 -->
    <arg name="allied_narrow_camera_auto_gain" default="false" />
    <arg name="allied_narrow_camera_auto_gain_range_min" default="0" /> <!-- 0 - 48 -->
    <arg name="allied_narrow_camera_auto_gain_range_max" default="48" /> <!-- 0 - 48 -->
    <arg name="allied_narrow_camera_gamma" default="1" /> <!-- 0.4 - 2.4 -->
    <arg name="allied_narrow_camera_saturation" default="1" /> <!-- 0 - 2 -->
    <arg name="allied_narrow_camera_sharpness" default="0" /> <!-- -12 - 12 -->
    <arg name="allied_narrow_camera_hue" default="0" /> <!-- -40 - 40 -->
    <arg name="allied_narrow_camera_intensity_auto_precedence" default="0" /> <!-- 0(MinimizeNoise) or 1(MinimizeBlur) -->
    <arg name="allied_narrow_camera_auto_white_balance" default="false" />
    <arg name="allied_narrow_camera_balance_ratio_selector" default="0" /> <!-- 0(Red), 1(Blue) -->
    <arg name="allied_narrow_camera_balance_ratio" default="2.4" /> <!-- 0 - 8 -->
    <arg name="allied_narrow_camera_balance_white_auto_rate" default="100" /> <!-- 0 - 100 -->
    <arg name="allied_narrow_camera_balance_white_auto_tolerance" default="5" /> <!-- 0 - 50 -->
    <arg name="allied_narrow_camera_auto_mode_region_height" default="2056" /> <!-- 0 - 2056 -->
    <arg name="allied_narrow_camera_auto_mode_region_width" default="2464" /> <!-- 0 - 2464 -->
    <arg name="allied_narrow_camera_intensity_controller_region" default="0" /> <!-- 0(AutoMode), 4(FullImage) -->
    <arg name="allied_narrow_camera_intensity_controller_target" default="50" /> <!-- 10 - 90 -->
    <arg name="allied_narrow_camera_max_driver_buffers_count" default="64" /> <!-- 1 - 4096 -->

    <!-- Start the driver -->
    <node pkg="l3cam_ros2" exec="l3cam_ros2_node" name="l3cam_ros2_node" output="screen">
        <param name="local_address" value="$(var local_address)" type="str" />
        <param name="device_address" value="$(var device_address)" type="str" />

        <param name="pointcloud_color" value="$(var pointcloud_color)" type="int" />
        <param name="pointcloud_color_range_minimum" value="$(var pointcloud_color_range_minimum)" type="int" />
        <param name="pointcloud_color_range_maximum" value="$(var pointcloud_color_range_maximum)" type="int" />
        <param name="distance_range_minimum" value="$(var distance_range_minimum)" type="int" />
        <param name="distance_range_maximum" value="$(var distance_range_maximum)" type="int" />
        <param name="auto_bias" value="$(var auto_bias)" type="bool" />
        <param name="bias_value_right" value="$(var bias_value_right)" type="int" />
        <param name="bias_value_left" value="$(var bias_value_left)" type="int" />

        <param name="polarimetric_camera_brightness" value="$(var polarimetric_camera_brightness)" type="int" />
        <param name="polarimetric_camera_black_level" value="$(var polarimetric_camera_black_level)" type="float" />
        <param name="polarimetric_camera_auto_gain" value="$(var polarimetric_camera_auto_gain)" type="bool" />
        <param name="polarimetric_camera_auto_gain_range_minimum" value="$(var polarimetric_camera_auto_gain_range_minimum)" type="float" />
        <param name="polarimetric_camera_auto_gain_range_maximum" value="$(var polarimetric_camera_auto_gain_range_maximum)" type="float" />
        <param name="polarimetric_camera_gain" value="$(var polarimetric_camera_gain)" type="float" />
        <param name="polarimetric_camera_auto_exposure_time" value="$(var polarimetric_camera_auto_exposure_time)" type="bool" />
        <param name="polarimetric_camera_auto_exposure_time_range_minimum" value="$(var polarimetric_camera_auto_exposure_time_range_minimum)" type="float" />
        <param name="polarimetric_camera_auto_exposure_time_range_maximum" value="$(var polarimetric_camera_auto_exposure_time_range_maximum)" type="float" />
        <param name="polarimetric_camera_exposure_time" value="$(var polarimetric_camera_exposure_time)" type="float" />

        <param name="rgb_camera_brightness" value="$(var rgb_camera_brightness)" type="int" />
        <param name="rgb_camera_contrast" value="$(var rgb_camera_contrast)" type="int" />
        <param name="rgb_camera_saturation" value="$(var rgb_camera_saturation)" type="int" />
        <param name="rgb_camera_sharpness" value="$(var rgb_camera_sharpness)" type="int" />
        <param name="rgb_camera_gamma" value="$(var rgb_camera_gamma)" type="int" />
        <param name="rgb_camera_gain" value="$(var rgb_camera_gain)" type="int" />
        <param name="rgb_camera_auto_white_balance" value="$(var rgb_camera_auto_white_balance)" type="bool" />
        <param name="rgb_camera_white_balance" value="$(var rgb_camera_white_balance)" type="int" />
        <param name="rgb_camera_auto_exposure_time" value="$(var rgb_camera_auto_exposure_time)" type="bool" />
        <param name="rgb_camera_exposure_time" value="$(var rgb_camera_exposure_time)" type="int" />
        <param name="rgb_camera_resolution" value="$(var rgb_camera_resolution)" type="int" />
        <param name="rgb_camera_framerate" value="$(var rgb_camera_framerate)" type="int" />

        <param name="thermal_camera_colormap" value="$(var thermal_camera_colormap)" type="int" />
        <param name="thermal_camera_temperature_filter" value="$(var thermal_camera_temperature_filter)" type="bool" />
        <param name="thermal_camera_temperature_filter_min" value="$(var thermal_camera_temperature_filter_min)" type="int" />
        <param name="thermal_camera_temperature_filter_max" value="$(var thermal_camera_temperature_filter_max)" type="int" />
        
        <param name="allied_wide_camera_black_level" value="$(var allied_wide_camera_black_level)" type="float" />
        <param name="allied_wide_camera_exposure_time" value="$(var allied_wide_camera_exposure_time)" type="float" />
        <param name="allied_wide_camera_auto_exposure_time" value="$(var allied_wide_camera_auto_exposure_time)" type="bool" />
        <param name="allied_wide_camera_auto_exposure_time_range_min" value="$(var allied_wide_camera_auto_exposure_time_range_min)" type="float" />
        <param name="allied_wide_camera_auto_exposure_time_range_max" value="$(var allied_wide_camera_auto_exposure_time_range_max)" type="float" />
        <param name="allied_wide_camera_gain" value="$(var allied_wide_camera_gain)" type="float" />
        <param name="allied_wide_camera_auto_gain" value="$(var allied_wide_camera_auto_gain)" type="bool" />
        <param name="allied_wide_camera_auto_gain_range_min" value="$(var allied_wide_camera_auto_gain_range_min)" type="float" />
        <param name="allied_wide_camera_auto_gain_range_max" value="$(var allied_wide_camera_auto_gain_range_max)" type="float" />
        <param name="allied_wide_camera_gamma" value="$(var allied_wide_camera_gamma)" type="float" />
        <param name="allied_wide_camera_saturation" value="$(var allied_wide_camera_saturation)" type="float" />
        <param name="allied_wide_camera_sharpness" value="$(var allied_wide_camera_sharpness)" type="float" />
        <param name="allied_wide_camera_hue" value="$(var allied_wide_camera_hue)" type="float" />
        <param name="allied_wide_camera_intensity_auto_precedence" value="$(var allied_wide_camera_intensity_auto_precedence)" type="int" />
        <param name="allied_wide_camera_auto_white_balance" value="$(var allied_wide_camera_auto_white_balance)" type="bool" />
        <param name="allied_wide_camera_balance_ratio_selector" value="$(var allied_wide_camera_balance_ratio_selector)" type="int" />
        <param name="allied_wide_camera_balance_ratio" value="$(var allied_wide_camera_balance_ratio)" type="float" />
        <param name="allied_wide_camera_balance_white_auto_rate" value="$(var allied_wide_camera_balance_white_auto_rate)" type="float" />
        <param name="allied_wide_camera_balance_white_auto_tolerance" value="$(var allied_wide_camera_balance_white_auto_tolerance)" type="float" />
        <param name="allied_wide_camera_auto_mode_region_height" value="$(var allied_wide_camera_auto_mode_region_height)" type="int" />
        <param name="allied_wide_camera_auto_mode_region_width" value="$(var allied_wide_camera_auto_mode_region_width)" type="int" />
        <param name="allied_wide_camera_intensity_controller_region" value="$(var allied_wide_camera_intensity_controller_region)" type="int" />
        <param name="allied_wide_camera_intensity_controller_target" value="$(var allied_wide_camera_intensity_controller_target)" type="float" />
        <param name="allied_wide_camera_max_driver_buffers_count" value="$(var allied_wide_camera_max_driver_buffers_count)" type="int" />

        <param name="allied_narrow_camera_black_level" value="$(var allied_narrow_camera_black_level)" type="float" />
        <param name="allied_narrow_camera_exposure_time" value="$(var allied_narrow_camera_exposure_time)" type="float" />
        <param name="allied_narrow_camera_auto_exposure_time" value="$(var allied_narrow_camera_auto_exposure_time)" type="bool" />
        <param name="allied_narrow_camera_auto_exposure_time_range_min" value="$(var allied_narrow_camera_auto_exposure_time_range_min)" type="float" />
        <param name="allied_narrow_camera_auto_exposure_time_range_max" value="$(var allied_narrow_camera_auto_exposure_time_range_max)" type="float" />
        <param name="allied_narrow_camera_gain" value="$(var allied_narrow_camera_gain)" type="float" />
        <param name="allied_narrow_camera_auto_gain" value="$(var allied_narrow_camera_auto_gain)" type="bool" />
        <param name="allied_narrow_camera_auto_gain_range_min" value="$(var allied_narrow_camera_auto_gain_range_min)" type="float" />
        <param name="allied_narrow_camera_auto_gain_range_max" value="$(var allied_narrow_camera_auto_gain_range_max)" type="float" />
        <param name="allied_narrow_camera_gamma" value="$(var allied_narrow_camera_gamma)" type="float" />
        <param name="allied_narrow_camera_saturation" value="$(var allied_narrow_camera_saturation)" type="float" />
        <param name="allied_narrow_camera_sharpness" value="$(var allied_narrow_camera_sharpness)" type="float" />
        <param name="allied_narrow_camera_hue" value="$(var allied_narrow_camera_hue)" type="float" />
        <param name="allied_narrow_camera_intensity_auto_precedence" value="$(var allied_narrow_camera_intensity_auto_precedence)" type="int" />
        <param name="allied_narrow_camera_auto_white_balance" value="$(var allied_narrow_camera_auto_white_balance)"  type="bool" />
        <param name="allied_narrow_camera_balance_ratio_selector" value="$(var allied_narrow_camera_balance_ratio_selector)" type="int" />
        <param name="allied_narrow_camera_balance_ratio" value="$(var allied_narrow_camera_balance_ratio)" type="float" />
        <param name="allied_narrow_camera_balance_white_auto_rate" value="$(var allied_narrow_camera_balance_white_auto_rate)" type="float" />
        <param name="allied_narrow_camera_balance_white_auto_tolerance" value="$(var allied_narrow_camera_balance_white_auto_tolerance)" type="float" />
        <param name="allied_narrow_camera_auto_mode_region_height" value="$(var allied_narrow_camera_auto_mode_region_height)" type="int" />
        <param name="allied_narrow_camera_auto_mode_region_width" value="$(var allied_narrow_camera_auto_mode_region_width)" type="int" />
        <param name="allied_narrow_camera_intensity_controller_region" value="$(var allied_narrow_camera_intensity_controller_region)" type="int" />
        <param name="allied_narrow_camera_intensity_controller_target" value="$(var allied_narrow_camera_intensity_controller_target)" type="float" />
        <param name="allied_narrow_camera_max_driver_buffers_count" value="$(var allied_narrow_camera_max_driver_buffers_count)" type="int" />
    </node>

    <!-- Start dyanmic reconfigure -->
    <include file="$(dirname)/configure_l3cam_launch.xml">
        <arg name="rqt_reconfigure" value="$(var rqt_reconfigure)" />

        <arg name="ip_address" value="$(var ip_address)" />
        <arg name="netmask" value="$(var netmask)" />
        <arg name="gateway" value="$(var gateway)" />
        <arg name="dhcp" value="$(var dhcp)" />

        <arg name="pointcloud_color" value="$(var pointcloud_color)" />
        <arg name="pointcloud_color_range_minimum" value="$(var pointcloud_color_range_minimum)" />
        <arg name="pointcloud_color_range_maximum" value="$(var pointcloud_color_range_maximum)" />
        <arg name="distance_range_minimum" value="$(var distance_range_minimum)" />
        <arg name="distance_range_maximum" value="$(var distance_range_maximum)" />

        <arg name="polarimetric_camera_brightness" value="$(var polarimetric_camera_brightness)" />
        <arg name="polarimetric_camera_black_level" value="$(var polarimetric_camera_black_level)" />
        <arg name="polarimetric_camera_auto_gain" value="$(var polarimetric_camera_auto_gain)" />
        <arg name="polarimetric_camera_auto_gain_range_minimum" value="$(var polarimetric_camera_auto_gain_range_minimum)" />
        <arg name="polarimetric_camera_auto_gain_range_maximum" value="$(var polarimetric_camera_auto_gain_range_maximum)" />
        <arg name="polarimetric_camera_gain" value="$(var polarimetric_camera_gain)" />
        <arg name="polarimetric_camera_auto_exposure_time" value="$(var polarimetric_camera_auto_exposure_time)" />
        <arg name="polarimetric_camera_auto_exposure_time_range_minimum" value="$(var polarimetric_camera_auto_exposure_time_range_minimum)" />
        <arg name="polarimetric_camera_auto_exposure_time_range_maximum" value="$(var polarimetric_camera_auto_exposure_time_range_maximum)" />
        <arg name="polarimetric_camera_exposure_time" value="$(var polarimetric_camera_exposure_time)" />

        <arg name="rgb_camera_brightness" value="$(var rgb_camera_brightness)" />
        <arg name="rgb_camera_contrast" value="$(var rgb_camera_contrast)" />
        <arg name="rgb_camera_saturation" value="$(var rgb_camera_saturation)" />
        <arg name="rgb_camera_sharpness" value="$(var rgb_camera_sharpness)" />
        <arg name="rgb_camera_gamma" value="$(var rgb_camera_gamma)" />
        <arg name="rgb_camera_gain" value="$(var rgb_camera_gain)" />
        <arg name="rgb_camera_auto_white_balance" value="$(var rgb_camera_auto_white_balance)" />
        <arg name="rgb_camera_white_balance" value="$(var rgb_camera_white_balance)" />
        <arg name="rgb_camera_auto_exposure_time" value="$(var rgb_camera_auto_exposure_time)" />
        <arg name="rgb_camera_exposure_time" value="$(var rgb_camera_exposure_time)" />

        <arg name="thermal_camera_colormap" value="$(var thermal_camera_colormap)" />
        <arg name="thermal_camera_temperature_filter" value="$(var thermal_camera_temperature_filter)" />
        <arg name="thermal_camera_temperature_filter_min" value="$(var thermal_camera_temperature_filter_min)" />
        <arg name="thermal_camera_temperature_filter_max" value="$(var thermal_camera_temperature_filter_max)" />

        <arg name="allied_wide_camera_exposure_time" value="$(var allied_wide_camera_exposure_time)" />
        <arg name="allied_wide_camera_auto_exposure_time" value="$(var allied_wide_camera_auto_exposure_time)" />
        <arg name="allied_wide_camera_auto_exposure_time_range_min" value="$(var allied_wide_camera_auto_exposure_time_range_min)" />
        <arg name="allied_wide_camera_auto_exposure_time_range_max" value="$(var allied_wide_camera_auto_exposure_time_range_max)" />
        <arg name="allied_wide_camera_gain" value="$(var allied_wide_camera_gain)" />
        <arg name="allied_wide_camera_auto_gain" value="$(var allied_wide_camera_auto_gain)" />
        <arg name="allied_wide_camera_auto_gain_range_min" value="$(var allied_wide_camera_auto_gain_range_min)" />
        <arg name="allied_wide_camera_auto_gain_range_max" value="$(var allied_wide_camera_auto_gain_range_max)" />
        <arg name="allied_wide_camera_gamma" value="$(var allied_wide_camera_gamma)" />
        <arg name="allied_wide_camera_saturation" value="$(var allied_wide_camera_saturation)" />
        <arg name="allied_wide_camera_hue" value="$(var allied_wide_camera_hue)" />
        <arg name="allied_wide_camera_intensity_auto_precedence" value="$(var allied_wide_camera_intensity_auto_precedence)" />
        <arg name="allied_wide_camera_auto_white_balance" value="$(var allied_wide_camera_auto_white_balance)" />
        <arg name="allied_wide_camera_balance_ratio_selector" value="$(var allied_wide_camera_balance_ratio_selector)" />
        <arg name="allied_wide_camera_balance_ratio" value="$(var allied_wide_camera_balance_ratio)" />
        <arg name="allied_wide_camera_balance_white_auto_rate" value="$(var allied_wide_camera_balance_white_auto_rate)" />
        <arg name="allied_wide_camera_balance_white_auto_tolerance" value="$(var allied_wide_camera_balance_white_auto_tolerance)" />
        <arg name="allied_wide_camera_intensity_controller_region" value="$(var allied_wide_camera_intensity_controller_region)" />
        <arg name="allied_wide_camera_intensity_controller_target" value="$(var allied_wide_camera_intensity_controller_target)" />

        <arg name="allied_narrow_camera_exposure_time" value="$(var allied_narrow_camera_exposure_time)" />
        <arg name="allied_narrow_camera_auto_exposure_time" value="$(var allied_narrow_camera_auto_exposure_time)" />
        <arg name="allied_narrow_camera_auto_exposure_time_range_min" value="$(var allied_narrow_camera_auto_exposure_time_range_min)" />
        <arg name="allied_narrow_camera_auto_exposure_time_range_max" value="$(var allied_narrow_camera_auto_exposure_time_range_max)" />
        <arg name="allied_narrow_camera_gain" value="$(var allied_narrow_camera_gain)" />
        <arg name="allied_narrow_camera_auto_gain" value="$(var allied_narrow_camera_auto_gain)" />
        <arg name="allied_narrow_camera_auto_gain_range_min" value="$(var allied_narrow_camera_auto_gain_range_min)" />
        <arg name="allied_narrow_camera_auto_gain_range_max" value="$(var allied_narrow_camera_auto_gain_range_max)" />
        <arg name="allied_narrow_camera_gamma" value="$(var allied_narrow_camera_gamma)" />
        <arg name="allied_narrow_camera_saturation" value="$(var allied_narrow_camera_saturation)" />
        <arg name="allied_narrow_camera_hue" value="$(var allied_narrow_camera_hue)" />
        <arg name="allied_narrow_camera_intensity_auto_precedence" value="$(var allied_narrow_camera_intensity_auto_precedence)" />
        <arg name="allied_narrow_camera_auto_white_balance" value="$(var allied_narrow_camera_auto_white_balance)" />
        <arg name="allied_narrow_camera_balance_ratio_selector" value="$(var allied_narrow_camera_balance_ratio_selector)" />
        <arg name="allied_narrow_camera_balance_ratio" value="$(var allied_narrow_camera_balance_ratio)" />
        <arg name="allied_narrow_camera_balance_white_auto_rate" value="$(var allied_narrow_camera_balance_white_auto_rate)" />
        <arg name="allied_narrow_camera_balance_white_auto_tolerance" value="$(var allied_narrow_camera_balance_white_auto_tolerance)" />
        <arg name="allied_narrow_camera_intensity_controller_region" value="$(var allied_narrow_camera_intensity_controller_region)" />
        <arg name="allied_narrow_camera_intensity_controller_target" value="$(var allied_narrow_camera_intensity_controller_target)" />
    </include>

    <!-- Start streaming the available sensors. -->
    <include file="$(dirname)/stream_l3cam_launch.xml">
        <arg name="rviz2" value="$(var rviz2)" />
    </include>

    <!-- The nodes that don't have their sensor available will automatically shut down -->
</launch>